guide/src/class/protocols.md

```diff
@@ -21,22 +21,40 @@ The magic methods handled by PyO3 are very similar to the standard Python ones o
  - Magic methods for the sequence protocol
 
 When PyO3 handles a magic method, a couple of changes apply compared to other `#[pymethods]`:
-  - The `#[pyo3(text_signature = "...")]` attribute is not allowed
-  - The types of the arguments are fixed according to the magic method
+ - The `#[pyo3(text_signature = "...")]` attribute is not allowed
+ - The signature is restricted to match the magic method
+
+The following sections list of all magic methods PyO3 currently handles.  The
+given signatures should be interpreted as follows:
+ - All methods take a receiver as first argument, which can be `&self`, `&mut
+   self` or a `PyCell` reference like `self_: PyRef<Self>` and `self_:
+   PyRefMut<Self>`, as described [here](../class.md#inheritance).
+ - An optional `Python<'py>` argument is always allowed as the first argument.
```

Comment:
I think in the way currently implemented `Python` arguments can be in any position. I'd be happy to restrict this, however.

Reply:
ah ok, I didn't even test this. I suggest keeping it like this in the docs anyway.